### ISSUE TITLE
Cassandra Request timeout in Hawkular cannot be changed from default value.

### ISSUE THREAD

**Issue by AlbertoPeon** [openshift/origin-metrics#413]
We are using image version `v3.7.1` for all cassandra-hawkular-1, hawkular-metrics and heapster.

We have recently seen an issue where Hawkular was timing out to create tables in Cassandra. The reason why Cassandra is slow to create the tables is understood in our side (seems the storage we use is just slow).

However, we would like to increase the request timeout to verify how long actually takes for all the table creation.

We suspect that we need to change the following variable in hawkular (ref. https://github.com/hawkular/hawkular-metrics/blob/4af918c69102e4d9b97743212c47d340734fe182/api/metrics-api-util/src/main/java/org/hawkular/metrics/api/jaxrs/config/ConfigurationKey.java#L54) as we see in the logs the timeout occurs exactly after 12 seconds.

However, we have tried to pass it in the hawkular entrypoint with:
```
-Dhawkular.metrics.cassandra.request-timeout=24000'
```

This seems to be taken into accoun as hawkular lists all the arguments on startup and I can see:

```
hawkular.metrics.cassandra.request-timeout = 24000
``` 
in the argument list.

However, this does not change anything and table creation keeps failing after 12 seconds. I have also tried to pass the `CASSANDRA_REQUEST_TIMEOUT` environment variable without success.

Could you suggest if it is possible to change that timeout and, if so, how we can do it?

Thank you very much!

**Comment by jsanda** [comment]
There are two timeouts of which you need to be aware. First, is the timeout in Cassandra which I believe defaults to 10 seconds. Let's say we have two cassandra pods - A and B. The client (which would be hawkular-metrics) sends a request to A which in turns routes it to B. If B does not respond to A within 10 seconds, then A will return a timeout exception to the client.

Secondly, there is a client-side timeout within the Cassandra driver. Setting the ```hawkular.metrics.cassandra.request-timeout``` system property configures this timeout setting. The different with the first timeout setting may not be immediately obvious. The driver setting applies to a request to a single Cassandra node. If you only ever run a single Cassandra pod then it does not make much sense, but it does when you have a multi-node Cassandra cluster. A single request can fan out to multiple Cassandra nodes, and this is where the driver timeout comes into play.

The timeout setting within Cassandra is configured within cassandra.yaml which is part of the Cassandra image. Unfortunately, it is not editable; however, there is [open ticket](https://bugzilla.redhat.com/show_bug.cgi?id=1539833) to change that.

Were the timeouts occurring at start up? Can you provide the logs for the cassandra and hawkular-metrics pods?

If you have a slow storage issue, then I recommend that you address it. Increasing timeout values will likely only delay more problems.

**Comment by matihost** [comment]
yes it happens during Hawkular Metrics pod startup.

it happened to me as well with OKD 3.9.0 (actually v3.9 due to
```
# https://github.com/openshift/origin/issues/19440  - mage v3.9.0 is not present in docker.io
# workaround : openshift_metrics_image_version has to be overridden
openshift_metrics_image_version=v3.9
```

The [ticket](https://bugzilla.redhat.com/show_bug.cgi?id=1539833) you mentioned is closed wontfix - w/o any explanation. Could you mention why? ( I foresee the reason is that Hawkular+Cassandra vanish in 3.11) ?

Relevant startup logs from Hawkular pod with timeout:
```
2018-09-01 11:57:53,609 INFO  [com.datastax.driver.core.Cluster] (ServerService Thread Pool -- 86) New Cassandra host hawkular-cassandra/172.30.30.128:9042 added
2018-09-01 11:57:53,999 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating type aggregation_template
2018-09-01 11:57:55,987 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table tenants_by_time
2018-09-01 11:57:58,570 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating type aggregate_data
2018-09-01 11:57:58,613 INFO  [org.hawkular.openshift.auth.org.hawkular.openshift.namespace.NamespaceHandler] (ServerService Thread Pool -- 86) Received namespace overrides: {}
2018-09-01 11:57:58,810 INFO  [org.hawkular.metrics.api.jaxrs.HawkularMetricsRestApp] (ServerService Thread Pool -- 86) HAWKMETRICS200001: Hawkular Metrics starting
2018-09-01 11:57:58,810 INFO  [org.jboss.resteasy.resteasy_jaxrs.i18n] (ServerService Thread Pool -- 86) RESTEASY002225: Deploying javax.ws.rs.core.Application: class org.hawkular.metrics.api.jaxrs.HawkularMetricsRestApp
2018-09-01 11:58:00,330 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table data
2018-09-01 11:58:01,237 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 86) WFLYUT0021: Registered web context: /hawkular/metrics
2018-09-01 11:58:01,824 INFO  [org.hawkular.metrics.alerter.groups.GroupTriggerManager] (ServerService Thread Pool -- 86) Starting Hawkular Metrics Group Trigger Manager, distributed=true
2018-09-01 11:58:01,837 INFO  [org.hawkular.metrics.alerter.ConditionManager] (ServerService Thread Pool -- 87) Starting Hawkular Metrics External Alerter, distributed=true
2018-09-01 11:58:02,078 INFO  [org.hawkular.metrics.alerter.ConditionManager] (ServerService Thread Pool -- 87) Registering Trigger UPDATE/REMOVE listener
2018-09-01 11:58:02,102 INFO  [org.hawkular.metrics.alerter.groups.GroupTriggerManager] (ServerService Thread Pool -- 86) Registering Trigger UPDATE/REMOVE listener
2018-09-01 11:58:02,354 INFO  [javax.enterprise.resource.webcontainer.jsf.config] (ServerService Thread Pool -- 87) Initializing Mojarra 2.2.13.SP1 20160303-1204 for context '/hawkular/__alerter-metrics'
2018-09-01 11:58:03,807 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table metrics_tags_idx
2018-09-01 11:58:03,928 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 87) WFLYUT0021: Registered web context: /hawkular/__alerter-metrics
2018-09-01 11:58:04,246 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 37) WFLYSRV0010: Deployed "hawkular-metrics.ear" (runtime-name : "hawkular-metrics.ear")
2018-09-01 11:58:06,053 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://127.0.0.1:9990/management
2018-09-01 11:58:06,054 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://127.0.0.1:9990
2018-09-01 11:58:06,069 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.1.0.Final (WildFly Core 2.2.0.Final) started in 120334ms - Started 1639 of 2091 services (849 services are lazy, passive or on-demand)
2018-09-01 11:58:09,777 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table metrics_idx
2018-09-01 11:58:14,774 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table retentions_idx
2018-09-01 11:58:18,650 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating type trigger_def
2018-09-01 11:58:19,919 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table tasks
2018-09-01 11:58:21,374 INFO  [org.hawkular.metrics.schema.SchemaService] (metricsservice-lifecycle-thread) Creating table leases
2018-09-01 11:58:23,927 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Creating change log table hawkular_metrics.cassalog
2018-09-01 11:58:27,170 WARN  [com.datastax.driver.core.Cluster] (cluster3-worker-0) Re-preparing already prepared query is generally an anti-pattern and will likely affect performance. Consider preparing the statement only once. Query='UPDATE hawkular_metrics.cassalog SET applied_at = ? WHERE bucket = ? AND revision = ?'
2018-09-01 11:58:27,323 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: set-keyspace
USE hawkular_metrics
--
2018-09-01 11:58:27,550 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 1.0

CREATE TABLE sys_config (
    config_id text,
    name text,
    value text,
    PRIMARY KEY (config_id, name)
) WITH compaction = { 'class': 'LeveledCompactionStrategy' }

--
2018-09-01 11:58:30,257 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 1.1
ALTER TABLE data WITH gc_grace_seconds = 86400
--
2018-09-01 11:58:31,510 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 1.2
ALTER TABLE data ADD tags map<text,text>
--
2018-09-01 11:58:32,979 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 1.3
ALTER TABLE data ADD s_value text
--
2018-09-01 11:58:35,914 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 1.4
INSERT INTO sys_config (config_id, name, value) VALUES ('org.hawkular.metrics', 'string-size', '2048')
--
2018-09-01 11:58:36,001 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: set-keyspace
USE hawkular_metrics
--
2018-09-01 11:58:36,043 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 2.0

CREATE TABLE locks (
    name text PRIMARY KEY,
    value text,
) WITH compaction = { 'class': 'LeveledCompactionStrategy' }

--
2018-09-01 11:58:37,665 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 2.1

CREATE TABLE jobs (
    id uuid PRIMARY KEY,
    type text,
    name text,
    params map<text, text>,
    trigger frozen <trigger_def>
) WITH compaction = { 'class': 'LeveledCompactionStrategy' }

--
2018-09-01 11:58:39,331 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 2.2

CREATE TABLE scheduled_jobs_idx (
    time_slice timestamp,
    job_id uuid,
    PRIMARY KEY (time_slice, job_id)
) WITH compaction = { 'class': 'LeveledCompactionStrategy' }

--
2018-09-01 11:58:41,030 INFO  [org.cassalog.core.CassalogImpl] (metricsservice-lifecycle-thread) Applying ChangeSet
-- version: 2.3

CREATE TABLE finished_jobs_idx (
    time_slice timestamp,
    job_id uuid,
    PRIMARY KEY (time_slice, job_id)
) WITH compaction = { 'class': 'LeveledCompactionStrategy' }

--
2018-09-01 11:59:01,159 FATAL [org.hawkular.metrics.api.jaxrs.MetricsServiceLifecycle] (metricsservice-lifecycle-thread) HAWKMETRICS200006: An error occurred trying to connect to the Cassandra cluster: java.lang.RuntimeException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: hawkular-cassandra/172.30.30.128:9042 (com.datastax.driver.core.exceptions.OperationTimedOutException: [hawkular-cassandra/172.30.30.128:9042] Timed out waiting for server response))
	at org.hawkular.metrics.api.jaxrs.DistributedLock.lockAndThen(DistributedLock.java:111)
	at org.hawkular.metrics.api.jaxrs.MetricsServiceLifecycle.initSchema(MetricsServiceLifecycle.java:667)
	at org.hawkular.metrics.api.jaxrs.MetricsServiceLifecycle.startMetricsService(MetricsServiceLifecycle.java:423)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: hawkular-cassandra/172.30.30.128:9042 (com.datastax.driver.core.exceptions.OperationTimedOutException: [hawkular-cassandra/172.30.30.128:9042] Timed out waiting for server response))
	at com.datastax.driver.core.exceptions.NoHostAvailableException.copy(NoHostAvailableException.java:84)
	at com.datastax.driver.core.exceptions.NoHostAvailableException.copy(NoHostAvailableException.java:37)
	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
	at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:245)
	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:68)
	at com.datastax.driver.core.Session$execute$0.call(Unknown Source)
	at org.cassalog.core.CassalogImpl.executeCQL(CassalogImpl.groovy:351)
	at sun.reflect.GeneratedMethodAccessor68.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:384)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1021)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:69)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:166)
	at org.cassalog.core.CassalogImpl$_applyChangeSet_closure16.doCall(CassalogImpl.groovy:323)
	at sun.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1021)
	at groovy.lang.Closure.call(Closure.java:426)
	at groovy.lang.Closure.call(Closure.java:442)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2030)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2015)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2056)
	at org.codehaus.groovy.runtime.dgm$162.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:274)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.cassalog.core.CassalogImpl.applyChangeSet(CassalogImpl.groovy:323)
	at sun.reflect.GeneratedMethodAccessor69.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:384)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1021)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:69)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
	at org.cassalog.core.CassalogImpl$_execute_closure3.doCall(CassalogImpl.groovy:135)
	at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1021)
	at groovy.lang.Closure.call(Closure.java:426)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.eachWithIndex(DefaultGroovyMethods.java:1946)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.eachWithIndex(DefaultGroovyMethods.java:1926)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.eachWithIndex(DefaultGroovyMethods.java:1976)
	at org.codehaus.groovy.runtime.dgm$174.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:274)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.cassalog.core.CassalogImpl.execute(CassalogImpl.groovy:109)
	at org.hawkular.metrics.schema.SchemaService.run(SchemaService.java:67)
	at org.hawkular.metrics.api.jaxrs.MetricsServiceLifecycle.lambda$initSchema$2(MetricsServiceLifecycle.java:669)
	at org.hawkular.metrics.api.jaxrs.DistributedLock.lockAndThen(DistributedLock.java:109)
	... 9 more
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: hawkular-cassandra/172.30.30.128:9042 (com.datastax.driver.core.exceptions.OperationTimedOutException: [hawkular-cassandra/172.30.30.128:9042] Timed out waiting for server response))
	at com.datastax.driver.core.RequestHandler.reportNoMoreHosts(RequestHandler.java:220)
	at com.datastax.driver.core.RequestHandler.access$1200(RequestHandler.java:50)
	at com.datastax.driver.core.RequestHandler$SpeculativeExecution.findNextHostAndQuery(RequestHandler.java:291)
	at com.datastax.driver.core.RequestHandler$SpeculativeExecution.retry(RequestHandler.java:459)
	at com.datastax.driver.core.RequestHandler$SpeculativeExecution.processRetryDecision(RequestHandler.java:437)
	at com.datastax.driver.core.RequestHandler$SpeculativeExecution.onTimeout(RequestHandler.java:796)
	at com.datastax.driver.core.Connection$ResponseHandler$1.run(Connection.java:1382)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:663)
	at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:738)
	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:466)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	... 1 more

*** JBossAS process (265) received TERM signal ***
2018-09-01 11:59:55,361 INFO  [org.jboss.as.server] (Thread-2) WFLYSRV0220: Server shutdown has been requested via an OS signal
2018-09-01 11:59:56,086 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-3) ISPN000080: Disconnecting JGroups channel server
2018-09-01 11:59:56,086 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-3) ISPN000082: Stopping the RpcDispatcher for channel server
2018-09-01 11:59:56,091 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-1) ISPN000080: Disconnecting JGroups channel hibernate
2018-09-01 11:59:56,091 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-1) ISPN000082: Stopping the RpcDispatcher for channel hibernate
2018-09-01 11:59:56,107 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-2) ISPN000080: Disconnecting JGroups channel web
2018-09-01 11:59:56,110 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-2) ISPN000082: Stopping the RpcDispatcher for channel web
2018-09-01 11:59:56,306 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 95) WFLYUT0022: Unregistered web context: /hawkular/__alerter-metrics
2018-09-01 11:59:57,108 INFO  [org.hawkular.metrics.alerter.groups.GroupTriggerManager] (ServerService Thread Pool -- 95) Stopping Hawkular Metrics Group Trigger Manager, distributed=true
2018-09-01 11:59:57,199 INFO  [org.hawkular.metrics.alerter.ConditionManager] (ServerService Thread Pool -- 89) Stopping Hawkular Metrics External Alerter, distributed=true
2018-09-01 11:59:57,407 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 89) WFLYUT0022: Unregistered web context: /hawkular/metrics
2018-09-01 11:59:57,850 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 90) WFLYUT0022: Unregistered web context: /hawkular/__alerter-elasticsearch
2018-09-01 11:59:58,130 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 95) WFLYCLINF0003: Stopped locks cache from hawkular-metrics container
2018-09-01 11:59:58,255 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-2) ISPN000080: Disconnecting JGroups channel hawkular-metrics
2018-09-01 11:59:58,256 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-2) ISPN000082: Stopping the RpcDispatcher for channel hawkular-metrics
2018-09-01 11:59:58,360 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 93) WFLYUT0022: Unregistered web context: /hawkular/actions/webhook
2018-09-01 11:59:58,704 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 89) WFLYUT0022: Unregistered web context: /hawkular/actions/email
2018-09-01 11:59:58,799 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 96) WFLYUT0022: Unregistered web context: /hawkular/actions/elasticsearch
2018-09-01 11:59:58,958 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 89) WFLYUT0022: Unregistered web context: /hawkular/alerts
2018-09-01 11:59:59,035 INFO  [org.hawkular.alerts.engine.impl.CassCluster] (ServerService Thread Pool -- 90) Closing Cassandra cluster session
```
